### ISSUE TITLE
Issue with "Search with a Query Task" select element

### ISSUE THREAD

**Issue by banuelosj** [Esri/geodev-hackerlabs#218]
The select element does not change the query when the user selects a different TYPE. It stays at the default 'MAX' query. The console outputs an error message "Cannot read property 'on' of null. I printed to the console the dom.byId("queryDiv"), and it printed null. I attached a screenshot of the error message.
![console_error_query_task](https://user-images.githubusercontent.com/36280386/36229299-9e4780d6-118b-11e8-9844-b39ee1b51bca.jpg)

**Comment by jgravois** [comment]
interesting. the chrome only error can be resolved by wiring up the event listener _before_ the API is used to place the dropdown on the page.

```js
// Select a sql query
on(dom.byId("queryDiv"), "change", function(e) {
  var sql = e.target.value;
  getFeatures(sql);
})

// Add select element to UI
view.ui.add(dom.byId("queryDiv"), {
  position: "top-right"
});
```

@banuelosj if you can you test this change on your side it'd be super helpful. bonus points if you can submit a PR afterward.

**Comment by banuelosj** [comment]
Yes it worked! I worked on it on my own and used a Promise to fix the error (I also switched the event listener so I guess the promise was not needed after all). Just switching the event listener before is a lot easier to understand than adding the Promise.

**Comment by banuelosj** [comment]
I will work on the PR right now
